src/drawer.canvasentry.js

```diff
@@ -171,12 +179,59 @@ export default class CanvasEntry {
      * @param {number} y Y start position
      * @param {number} width Width of the rectangle
      * @param {number} height Height of the rectangle
+     * @param {number} radius Radius of the rectangle
      */
-    fillRectToContext(ctx, x, y, width, height) {
+    fillRectToContext(ctx, x, y, width, height, radius) {
         if (!ctx) {
             return;
         }
-        ctx.fillRect(x, y, width, height);
+
+        if (radius) {
+            this.drawRoundedRectangle(ctx, x, y, width, height, radius);
+        } else {
+            ctx.fillRect(x, y, width, height);
+        }
+    }
+
+    /**
+     * Draw a rounded rectangle on Canvas
+     *
+     * @private
+     * @param {CanvasRenderingContext2D} ctx Canvas context
+     * @param {number} x X-position of the rectangle
+     * @param {number} y Y-position of the rectangle
+     * @param {number} width Width of the rectangle
+     * @param {number} height Height of the rectangle
+     * @param {number} radius Radius of the rectangle
+     *
+     * @return {void}
+     * @example drawRoundedRectangle(ctx, 50, 50, 5, 10, 20)
+     */
+    drawRoundedRectangle(ctx, x, y, width, height, radius) {
+        if (height === 0) {
+            return;
+        }
+        if (height < 0) {
```

Comment:
can you add a comment here, describing why this check is added?

Reply:
Hi @thijstriemstra , as I understand it, a waveform consists of a peaks with float values from -1 to 1. The peak value is used to calculate bar height: https://github.com/katspaugh/wavesurfer.js/blob/8ba3328102da9746ce1fd390c0801f01edaaf612/src/drawer.multicanvas.js#L295
That's why a bar height can be negative. We need to use absolute height value in order to correctly calculate rounded rectangle coordinates because the [formula](https://github.com/katspaugh/wavesurfer.js/issues/953#issuecomment-510230749) expects the height to be positive. Does that make sense?

Reply:
makes sense, a good comment to add there. I will add it and merge this PR, thanks.